util/bloom_test.cc

```diff
@@ -601,6 +603,255 @@ TEST_P(FullBloomTest, OptimizeForMemory) {
   }
 }
 
+TEST_P(FullBloomTest, ReserveBloomAndRibbonFilterConstructionMemory) {
```

Comment:
This seems like too much unit test for the job we are doing here. IMHO, regressions are at least as likely in the integration logic as they are in the code units tested here. Unit tests at this level of detail make code evolution substantially more difficult, so have to offer sufficient value to justify their existence. While this is an important feature, it is among the least "critical" that we write.

IMHO we can reasonably cover this feature with an integration test (as in table_test; see other comment on testing), and partitioned filter should just change the calculation on peak usage.